src/ucs/config/parser.c

```diff
@@ -2027,6 +2045,22 @@ ucs_config_parser_print_field(FILE *stream, const void *opts, const char *env_pr
     ucs_assert(!ucs_list_is_empty(prefix_list));
     head = ucs_list_head(prefix_list, ucs_config_parser_prefix_t, list);
 
+    if ( snprintf(name_buf, sizeof(name_buf), "%s%s%s", env_prefix, head->prefix, name) >= (int)(sizeof(name_buf))) {
+        ucs_error("filter buffer overrun -- please print config without filters");
+        return;
+    }
+
+    if (ucs_config_parser_is_filtered(name_buf, filters)) {
+        return;
+    }
+
+    if ((flags & UCS_CONFIG_PRINT_HEADER) && (*printed == 0)) {
```

Comment:
we could check ucs_config_parser_is_filtered before calling ucs_config_parser_print_field, so that ucs_config_parser_print_field would not have to be made responsible for printing the header

Reply:
What if ucs_config_parser_print_opts checked the filter before calling ucs_config_parser_print_field?

Reply:
Maybe instead of adding "printed" argument, we can pass along flags by pointer (flags_p), and when header is printed we can remove the bit UCS_CONFIG_PRINT_HEADER from flags?